narwhals/_dask/namespace.py

```diff
@@ -398,8 +398,15 @@ def __call__(self, df: DaskLazyFrame) -> Sequence[dx.Series]:
         plx = df.__narwhals_namespace__()
         condition = parse_into_expr(self._condition, namespace=plx)(df)[0]
         condition = cast("dx.Series", condition)
+
         try:
-            value_series = parse_into_expr(self._then_value, namespace=plx)(df)[0]
+            then_expr = parse_into_expr(self._then_value, namespace=plx)
+            value_series = then_expr(df)[0]
+            # literal or reduction case
+            if then_expr._returns_scalar:  # type: ignore[attr-defined]
+                _df = condition.to_frame("a")
+                _df["tmp"] = value_series[0]
+                value_series = _df["tmp"]
```

Comment:
this is too big to be in a `try` clause, it should either be in `else`, or outside of try-except